experimental/Schemes/Resolution_structure.jl

```diff
@@ -0,0 +1,738 @@
+## Warnung: show auf desingMor geht noch nicht!!!
+export _desing_curve
+export  find_refinement_with_local_system_of_params
+
+#####################################################################################################
+# Desingularization morphism: birational map between covered schemes with smooth domain
+#####################################################################################################
+
+# Fehlt: NormalizationMorphism fuer Schemata -- muessten wir haben, sobald wir Lipman machen wollen
+#
+#@attributes mutable struct LipmanStyleSequence{
+#    DomainType<:AbsCoveredScheme,
+#    CodomainType<:AbsCoveredScheme
+#   } <: AbsDesingMor{
+#                                 DomainType,
+#                                 CodomainType,
+#                    }
+#  maps::Vector{:<union{BlowupMorphism,NormalizationMorphism}}        # count right to left:
+#                                                 # original scheme is codomain of map 1
+#  # boolean flags
+#  resolves_sing::Bool                            # domain not smooth yet?
+#  is_trivial::Bool                               # codomain already smooth?
+#
+#  # fields for caching, may be filled during computation
+#  ex_div::Vector{<:EffectiveCartierDivisor}      # list of exc. divisors arising from individual steps
+                                                  # in domain(maps[end])
+#
+#  # fields for caching to be filled a posteriori (on demand, only if partial_res==false)
+#  composed_map::AbsCoveredSchemeMorphism        
+#  exceptional_divisor::WeilDivisor          
+#
+#  function LipmanStyleSequence(maps::Vector{<:AbsCoveredSchemeMorphism})
+#    n = length(maps)
+#    for i in 1:n-1
+#      @assert domain(maps[i]) === codomain(maps[i+1]) "not a sequence of morphisms"
+#    end
+#    return new{typeof(domain(maps[end])),typeof(codomain(first(maps)))}(maps)
+#  end
+#end
+
+
+##################################################################################################
+# getters
+##################################################################################################
+maps(phi::AbsDesingMor) = copy(phi.maps)
```

Comment:
Why copy? Typically our getters do not copy stuff. The user should assume that anything returned is unsafe to mutate?
... but it is a common pitfall

Reply:
@afkafkafk13 and I have talked about this. I suggested to return a copy. Otherwise, the chances are high that unsafe code leads to a mess with the internals of the structure. And this tends to be almost impossible to debug. The way to solve this, is to provide the getter with the index and encourage the user to use that instead of creating copies all over the place. 

And I could live with `morphisms` and `morphism` below.

Reply:
taking your other suggestions for `morphisms` instead of  `maps` into account, I would go for

> morphisms(phi::AbsDesingMor) = phi.maps
> morphisms_copy(phi::AbsDesingMor) = copy(phi.maps)
> morphism(phi::AbsDesingMor, i::Int) = phi.maps[i]

Do you agree? Note that morphisms_copy is necessary at certain places, because the desingularization has to modify it by appending. I will carefully go through the existing code and change to morphisms_copy precisely, where it is necessary.

Reply:
No `morphisms_copy`, please! `morphisms` should create a copy each time and not give access to the internals. Such things have proved to be devastating in many instances. IMHO saving memory allocations has to be achieved by other means and this is also possible, e.g. by encouraging the use of `morphism(phi, i)`.

Reply:
> Typically our getters do not copy stuff.

I do not think that this is true. Even `gens` for polynomial rings usually creates a copy. And it does so for good reasons.

Reply:
ok then just morphisms(phi) and morphism(phi,i) giving back copies. I will make the appropriate changes.

Reply:
best rename `.maps` to `.morphisms` as well for consistency.

Reply:
Regarding `copy` in `morphisms` (and similar elsewhere in `gens` etc.): for OSCAR 2.0 we could consider using a type like the one defined in https://github.com/bkamins/ReadOnlyArrays.jl which implements the `AbstractArray` / `AbstractVector` interface by wrapping an existing array/vector and passing all operations through -- except anything that *writes* to the vector. The whole thing is basically "free" in terms of overhead.

But the downside is that the result is not a `Vector` anymore -- code which accepts `AbstractVector` will keep working, but surely a lot of code will need (trivial) adjustments -- hence this can't go into Oscar 1.x


In this regard, it would be much better if `Array` just had a bit flag that turned any attempt to write to it into an error

Reply:
> best rename `.maps` to `.morphisms` as well for consistency.

That was part of the content of my last statement.